spec/java/util/stream/DoubleStream.main.lsl

```diff
@@ -0,0 +1,1266 @@
+        );
+
+        if (i < this.length)
+            result = true;
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    @Phantom proc _predicate_condition_increment_loop (i: int): void
+    {
+        i += 1;
+    }
+
+
+    fun *.allMatch (@target self: DoubleStream, predicate: DoublePredicate): boolean
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (predicate == null)
+            _throwNPE();
+
+        result = true;
+
+        if (this.length > 0)
+        {
+            result = false
+            var i: int = 0;
+            action LOOP_WHILE(
+                i < this.length && action CALL(predicate, [this.storage[i]]),
+                _predicate_condition_increment_loop(i)
+            );
+
+            if (i == this.length)
+                result = true;
+        }
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.noneMatch (@target self: DoubleStream, predicate: DoublePredicate): boolean
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (predicate == null)
+            _throwNPE();
+
+        result = true;
+
+        if (this.length > 0)
+        {
+            result = false
+            var i: int = 0;
+            action LOOP_WHILE(
+                i < this.length && !action CALL(predicate, [this.storage[i]]),
+                _predicate_condition_increment_loop(i)
+            );
+
+            if (i == this.length)
+                result = true;
+        }
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.findFirst (@target self: DoubleStream): OptionalDouble
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        result = _findFirst();
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.findAny (@target self: DoubleStream): OptionalDouble
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        result = _findFirst();
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.iterator (@target self: DoubleStream): PrimitiveIterator_OfDouble
+    {
+        result = new DoubleStreamIteratorAutomaton(state = Initialized,
+            parent = self,
+            cursor = 0,
+        );
+    }
+
+
+    fun *.spliterator (@target self: DoubleStream): Spliterator_OfDouble
+    {
+        action TODO();
+    }
+
+
+    // within java.util.stream.BaseStream
+    fun *.isParallel (@target self: DoubleStream): boolean
+    {
+        result = this.isParallel;
+    }
+
+
+    fun *.sequential (@target self: DoubleStream): DoubleStream
+    {
+        this.isParallel = false;
+        result = self;
+    }
+
+
+    fun *.parallel (@target self: DoubleStream): DoubleStream
+    {
+        this.isParallel = true;
+        result = self;
+    }
+
+
+    fun *.unordered (@target self: DoubleStream): DoubleStream
+    {
+        result = self;
+    }
+
+
+    // within java.util.stream.BaseStream
+    fun *.onClose (@target self: DoubleStream, arg0: Runnable): DoubleStream
```

Comment:
Consider adding a comment or `#todo` for handling exception handling and runnable chaining as it being done in the original (see `java.util.stream.AbstractPipeline#onClose`).

Reply:
Fixed;